### PR TITLE
GatherTree specification refactored

### PR DESCRIPTION
**Pull request by ggalieroc**
### Details:
 - [x] *Modified category section*
 - [x] *Detailed description improved*
 - [x] *Inputs/Outputs description improved*
 - [x] *Modified types section to supported types*

### Tickets:
 - *45850*

### CODE REVIEW

File: docs/ops/movement/GatherTree_1.md [hunk 1]
```diff
@@ -2,63 +2,60 @@
...
             if(finished):
-                final_idx[time, batch, beam] = end_token
-            elif(final_idx[time, batch, beam] == end_token):
+                final_ids[time, batch, beam] = end_token
+            elif(final_ids[time, batch, beam] == end_token):
                 finished = True
 ```
 
-Element data types for all input tensors should match each other.
+*GatherTree* operation is equivalent to [GatherTree operation in TensorFlow](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/gather_tree).
```
Comment (bszmelcz) [inline docs/ops/movement/GatherTree_1.md:40]: Shouldn't we avoid links to outer sources?

File: docs/ops/movement/GatherTree_1.md [hunk 2]
```diff
@@ -2,63 +2,60 @@
...
-
-* **3**:  `max_seq_len` -- a tensor of shape `[BATCH_SIZE]` of type *T* with maximum lengths for each sequence in the batch. **Required.**
-
-* **4**:  `end_token` -- a scalar tensor of type *T* with value of the end marker in a sequence. **Required.**
-
+* **1**:  `step_ids` - Indices from per each step. A tensor of type *T* and rank 3. Layout is `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]`. **Required.**
+* **2**:  `parent_ids` - Parent beam indices. A tensor of type *T* and rank3. Layout is `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]`. **Required.**
+* **3**:  `max_seq_len` - Maximum lengths for each sequence in the batch. A tensor of type *T* and rank 1. Layout is `[BATCH_SIZE]`. **Required.**
+* **4**:  `end_token` - Value of the end marker in a sequence. A scalar of type *T*. **Required.**
+* **Note**: Inputs should have integer values only.
```
Comment (mitruska) [inline docs/ops/movement/GatherTree_1.md:50]: As you changed the type *T* to any supported numeric type, this note seems quite confusing. What happens with float values? Are they trunc? Does it throw?;)

File: docs/ops/movement/GatherTree_1.md (hunk 1 above)
Comment (ggalieroc) [reply docs/ops/movement/GatherTree_1.md:40]: My understanding is that we should avoid to **ONLY** refer to 3rd party web-site with explanation. But the algorithm is shortly described and pseudocode provided above. I believe there is no harm on keeping this reference here.

File: docs/ops/movement/GatherTree_1.md (hunk 2 above)
Comment (ggalieroc) [reply docs/ops/movement/GatherTree_1.md:50]: This operation supports floating-point and integer precisions. As we agreed in guidelines for spec, we should avoid shorting precisions supported, and rather specify floating-point, integer, or if both, numeric. If in the future CPU plugin decides to support i8, spec would still be correct.

As for the Note section, the operation expects integer values, independently on the precision of the inputs.

File: docs/ops/movement/GatherTree_1.md [hunk 3]
```diff
@@ -2,63 +2,60 @@
...
 
-* **1**:  `step_ids` -- a tensor of shape `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]` of type *T* with indices from per each step. **Required.**
-
-* **2**:  `parent_idx` -- a tensor of shape `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]` of type *T* with parent beam indices. **Required.**
-
-* **3**:  `max_seq_len` -- a tensor of shape `[BATCH_SIZE]` of type *T* with maximum lengths for each sequence in the batch. **Required.**
-
-* **4**:  `end_token` -- a scalar tensor of type *T* with value of the end marker in a sequence. **Required.**
-
+* **1**:  `step_ids` - Indices from per each step. A tensor of type *T* and rank 3. Layout is `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]`. **Required.**
```
Comment (dkozykowski) [inline docs/ops/movement/GatherTree_1.md:46]: I think we could make it a bit more clear here in the first sentence.

**Review by bszmelcz — APPROVED** [review]
LGTM

File: docs/ops/movement/GatherTree_1.md [hunk 4]
```diff
@@ -2,63 +2,60 @@
...
+* **2**:  `parent_ids` - Parent beam indices. A tensor of type *T* and rank3. Layout is `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]`. **Required.**
```
[the remark this answers is not in the document]
Comment (ggalieroc) [reply docs/ops/movement/GatherTree_1.md]: done

File: docs/ops/movement/GatherTree_1.md (hunk 3 above)
Comment (ggalieroc) [reply docs/ops/movement/GatherTree_1.md]: done

**Comment by ilyachur** [comment]
@rkazants Please have a look

**Review by dkozykowski — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**